src/Spago/Command/Fetch.purs

```diff
@@ -541,6 +547,15 @@ getPackageDependencies packageName package = case package of
         , indent $ toDoc errLines
         ]
 
+resolvePackageVersionsToRanges :: Map PackageName Package -> Map PackageName Range -> Array VersionResolution
+resolvePackageVersionsToRanges registry = Map.toUnfoldable >>> Array.foldl
+  ( \acc (Tuple name requested) ->
+      case (Map.lookup name registry) of
+        Just (RegistryVersion resolved) -> Array.snoc acc { name, requested, resolved }
+        _ -> acc
+  )
+  []
```

Comment:
This can be condensed with `mapMaybeWithKey` (and let's move it to the `let` block as well):
```suggestion
resolvePackageVersionsToRanges registry = Array.fromFoldable
  <<< Map.values
  <<< Map.mapMaybeWithKey \name requested ->
    Map.lookup name registry <#> \(RegistryVersion resolved) ->
      { name, requested, resolved }
```